pkg/memory-monitor/src/monitor/cgroups.c

```diff
@@ -0,0 +1,332 @@
+// Copyright (c) 2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <errno.h>
+#include <fcntl.h>
+#include <limits.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/select.h>
+#include <syslog.h>
+#include <unistd.h>
+
+#include "config.h"
+#include "event.h"
+#include "monitor.h"
+#include "util.h"
+
+#include "cgroups.h"
+
+unsigned long cgroup_get_memory_usage(const char *cgroup_name) {
+    char path[PATH_MAX + 1];
+    int fd;
+    char str_usage[256];
+    unsigned long usage_bytes;
+
+    snprintf(path, sizeof(path), "%s/%s/memory.usage_in_bytes", CGROUP_PATH_PREFIX, cgroup_name);
+    fd = open(path, O_RDONLY);
+    if (fd == -1) {
+        syslog(LOG_ERR, "opening usage_in_bytes: %s", strerror(errno));
+        return 0;
+    }
+
+    ssize_t nread = read(fd, str_usage, sizeof(str_usage) - 1);
+    if (nread <= 0) {
+        syslog(LOG_ERR, "reading usage_in_bytes: %s", strerror(errno));
+        close(fd);
+        return 0;
+    }
+
+    str_usage[nread] = '\0';
+    bool error;
+    usage_bytes = strtoudec(str_usage, &error);
+    if (error) {
+        syslog(LOG_ERR, "strtoul: %s", strerror(errno));
+        close(fd);
+        return 0;
+    }
+
+    close(fd);
```

Comment:
Maybe we want to check close error code? This could be potential leak

Reply:
We can check. But what is the reaction in the case of error? Exit entire program? 
```
EBADF  fd isn't a valid open file descriptor.

EINTR  The close() call was interrupted by a signal; see
              signal(7).

EIO    An I/O error occurred.
``|

Reply:
I would log the fact that fd was not closed and continued work. I think exiting program would be too dramatic :)

Reply:
this is a read, who cares 😄 no cache or anything to be worried about.